ultimate_cron.module

```diff
@@ -1,76 +1,16 @@
 <?php
 /**
  * @file
- * Ultimate Cron. Extend cron functionality in Drupal.
+ * Ultimate Cron bootstrap file.
  */
 
-use Drupal\Core\Logger\RfcLogLevel;
-use Drupal\ultimate_cron\Logger\LoggerBase;
-use Drupal\ultimate_cron\CronRule;
 use Drupal\ultimate_cron\Entity\CronJob;
 
-//include dirname(__FILE__) . '/ultimate_cron.poorman.inc';
-
 define('ULTIMATE_CRON_LOG_TYPE_NORMAL', 0);
 define('ULTIMATE_CRON_LOG_TYPE_ADMIN', 1);
 define('ULTIMATE_CRON_LOG_TYPE_ALL', -1);
 
 /**
- * Pseudo define.
- */
-function _ultimate_cron_define_log_type_all() {
-  return array(ULTIMATE_CRON_LOG_TYPE_NORMAL, ULTIMATE_CRON_LOG_TYPE_ADMIN);
-}
-
-/**
- * Memory exhaust protection.
- * To reduce the risk of shutdown handlers failing, we reserve 2MiB of memory
- * by default. This value is overridable.
- */
-
-//$GLOBALS['__RESERVED_MEMORY'] = str_repeat('0', variable_get('ultimate_cron_reserve_memory', 1024 * 1024 * 2));
-$GLOBALS['ultimate_cron_shutdown_functions'] = array();
-
-/**
- * The shutdown function itself is also overridable, in case it is necessary to
- * add it earlier, say settings.php.
- * Remeber to invoke the registered ultimate cron shutdown handlers.
- * If the function exists, we assume that the register_shutdown_handler() has
- * also been setup correctly.
- */
-if (!function_exists('_ultimate_cron_out_of_memory_protection')) {
```

Comment:
leave this for now, add a task "Move _ultimate_cron_out_of_memory_protection()" to a service.